fem/libceed/ceed.hpp

```diff
@@ -197,6 +197,7 @@ void CeedAssembleDiagonalPA(const CeedData *ceedDataPtr,
 inline bool DeviceCanUseCeed()
 {
    return Device::Allows(Backend::CEED_CUDA) ||
+          Device::Allows(Backend::CEED_HIP) ||
```

Comment:
```suggestion
   return Device::Allows(Backend::DEVICE_MASK)
```

Reply:
In this one, do I really want to change it to `DEVICE_MASK`?  Isn't it specifically checking for `CEED_CUDA` or `CEED_HIP`, not any device usage?  Should I replace all three checks with `CEED_MASK` instead?

Reply:
Or wait, I see, the `CEED_CPU` check goes with the check on not allowing `DEVICE` or `OMP`.  So should there be a `CEED_DEVICE_MASK` to check for `CEED_CUDA` or `CEED_HIP` together?

Reply:
You're right in this one, `CEED_MASK` is the correct one. I don't really understand the logic for `(Device::Allows(Backend::CEED_CPU) && !Device::Allows(Backend::DEVICE_MASK|Backend::OMP_MASK));`, I think it's a bit overkill, we could probably replace everything with:
```c++
return Device::Allows(Backend::CEED_MASK);
```

Reply:
I did this in b267750431f09086aefa0494e2ab366ba2ef2620